java/test/jmri/jmrit/display/layoutEditor/PositionablePointTest.java

```diff
@@ -16,16 +21,381 @@
+        jmri.util.JUnitAppender.assertErrorMessage("Invalid connection type 0");
+
+        // test valid connection type
+        Assert.assertTrue("pp.isDisconnected(valid type) is null",
+                pp.isDisconnected(LayoutEditor.POS_POINT));
+    }
+
+    @Test
+    public void testSetTrackConnection() {
+        Assume.assumeFalse(GraphicsEnvironment.isHeadless());
+        Assert.assertNotNull("LayoutEditor exists", le);
+
+        PositionablePoint pp = new PositionablePoint("test", PositionablePoint.ANCHOR, MathUtil.zeroPoint2D, le);
+        Assert.assertNotNull("exists", pp);
+
+        // test null track segment
+        Assert.assertFalse("pp.setTrackConnection(null) is false",
+                pp.setTrackConnection(null));
+
+        PositionablePoint ppA = new PositionablePoint("A", PositionablePoint.ANCHOR, new Point2D.Double( 0.0,  0.0), le);
+        PositionablePoint ppB = new PositionablePoint("B", PositionablePoint.ANCHOR, new Point2D.Double(10.0, 10.0), le);
+        PositionablePoint ppC = new PositionablePoint("C", PositionablePoint.ANCHOR, new Point2D.Double(20.0, 20.0), le);
+        PositionablePoint ppD = new PositionablePoint("D", PositionablePoint.ANCHOR, new Point2D.Double(30.0, 30.0), le);
+        TrackSegment tsAB = new TrackSegment("testAB", ppA, LayoutTrack.POS_POINT, ppB, LayoutTrack.POS_POINT, false, false, le);
+        Assert.assertNotNull("Track Segment AB exists", tsAB);
+        TrackSegment tsBC = new TrackSegment("testBC", ppB, LayoutTrack.POS_POINT, ppC, LayoutTrack.POS_POINT, false, false, le);
+        Assert.assertNotNull("Track Segment BC exists", tsBC);
+        TrackSegment tsCD = new TrackSegment("testCD", ppC, LayoutTrack.POS_POINT, ppD, LayoutTrack.POS_POINT, false, false, le);
+        Assert.assertNotNull("Track Segment CD exists", tsCD);
+
+        // test non-null track segment
+        Assert.assertTrue("pp.setTrackConnection(tsAB) is true",
+                pp.setTrackConnection(tsAB));
+
+        // test already connected is true
+        Assert.assertTrue("pp.setTrackConnection(tsAB) is true",
+                pp.setTrackConnection(tsAB));
+
+        // test 2nd non-null track segment
+        Assert.assertTrue("pp.setTrackConnection(tsBC) is true",
+                pp.setTrackConnection(tsBC));
+
+        // test already connected is true
+        Assert.assertTrue("pp.setTrackConnection(tsBC) is true",
+                pp.setTrackConnection(tsBC));
+
+        // test 3rd non-null track segment
+        Assert.assertFalse("pp.setTrackConnection(tsCD) is true",
+                pp.setTrackConnection(tsCD));
+        jmri.util.JUnitAppender.assertErrorMessage("Attempt to assign more than allowed number of connections");
     }
 
     // from here down is testing infrastructure
+    @BeforeClass
+    public static void beforeClass() {
+        JUnitUtil.setUp();
+        if (!GraphicsEnvironment.isHeadless()) {
+            le = new LayoutEditor();
```

Comment:
If you want a new layout editor for each test, this should be in the @Before annotayed method; I highly recommend that since test order within a class is not guaranteed.

Reply:
The LayoutEditor has it's own set of tests; these tests are for PositionablePoint's only. AFAICT pp's only use the LE reference while drawing. IMHO it's not worth the additional overhead to create and destroy the LE reference for every test (when it's not what's being tested).

Reply:
Generally, you only want to put something in an @BeforeClass annotated method if it is very costly to create.  Something like creating a database would be considered a very costly operation as would opening connections to servers.

The biggest potential issue in this case really is that since tests MAY run in parallel, having a ( potentially ) long lived panel could result in a fragile test.  In particular, we have tests now that close all open windows and the LayoutEditor is a window, so it could be closed by another test.